### ISSUE TITLE
Investigate how to streamline association approval process in curation interface

### ISSUE THREAD

**Issue by eks-ebi** [EBISPOT/goci#211]
When a curator submits a template through the deposition interface and then import into the curation interface, nearly all of the data is added and validated automatically (assuming there are no errors). This includes automatic mapping of associations.

However, there is currently a requirement that curators mark each association in a study as "Approved" before the study can be published in the GWAS Catalog. Currently, the fastest way for curators to do this is to open each study individually and press an "Approve all associations" button. If multiple studies are imported, each with associations, it will be very time consuming to open and approve associations for each of these studies before publishing.

![Approve all assocs](https://user-images.githubusercontent.com/42406380/93459071-1741c780-f8d9-11ea-8436-59dd74e712e2.png)

**Task**

Investigate a more streamlined way for curators to approve associations for multiple studies in a publication at once.

**Comment by eks-ebi** [comment]
One idea could be to add an "Approve all associations" button to the publication-centric view (example location shown in red box). A curator could then select studies from the list (or select all studies by clicking on the box in the red circle), and click the button to approve all associations for the selected studies as a single action.

![Screenshot 2020-09-25 at 11.09.03.png](https://images.zenhubusercontent.com/5ebd0095cdca5a6323f696ae/50bf94b9-70ec-4c0b-8a63-d614aa1dec34)

**Comment by eks-ebi** [comment]
However, please also investigate if there are alternative ways to avoid curators having to click in and out of multiple studies to approve associations for each one by one.

**Comment by eks-ebi** [comment]
Looks good!

Tested with this publication: http://snoopy:9680/gwas/curation/publication/32804141
(has 4 studies, each with one association)

1. Selected all studies and pressed "Approve associations" - all associations were successful approved across the studies
2. Selected all studies and changed status to "Publish study" - worked (publishing was not blocked by unapproved associations)

**Comment by eks-ebi** [comment]
Also tested the opposite for this publication (i.e. tried to publish studies without first approving the associations): http://snoopy:9680/gwas/curation/publication/32527199
(two studies with 3 and 44 associations in each)

- Correctly prevented the studies from being published because the SNPs had not yet been approved.
- However there is no indication of this from the message displayed when you try to set status to "Publish study" (see screenshot below) - could we change this message to more accurately reflect the fact that the change of status has been blocked due to the presence of unapproved associations?

The alert suggests that something has been updated for the 2 studies, but actually there is no change to the Curation Status, which correctly stays as "Level 1 curation done":
![Screenshot 2020-12-07 at 13.25.00.png](https://images.zenhubusercontent.com/5ebd0095cdca5a6323f696ae/275ff869-62a8-46fe-8474-1e298fc50348)

**Comment by tudorgroza** [comment]
@eks-ebi - I'm afraid this second one is something else :) - it's in a different part of the code.
Is it ok if I move your last comment into a new ticket and we can move this one in Done if you're happy with the testing of the initial requirements?

**Comment by eks-ebi** [comment]
Yes absolutely! I'm happy with the initial requirements.

**Comment by tudorgroza** [comment]
Great! Thank you. Moved it here: https://github.com/EBISPOT/goci/issues/309

**Comment by buniello** [comment]
I agree with @eks-ebi on the need for clearer messages. Otherwise, it looks good!

**Comment by tudorgroza** [comment]
Thank you! Will consider this done and plan it for release on Thu.

**Comment by eks-ebi** [comment]
After further testing, a bug was encountered:

When a curator clicks “Approve assocations” on e.g. this page https://www.ebi.ac.uk/gwas/curation/publication/29874175, it should set all the associations in the selected studies to “Approved”. This is required before the curator can change Curator Status to “Publish study” for those studies.

Sometimes the "Approve associations" button sets associations to "Approved" correctly, other times the action fails without giving an error message. But it seems to be unpredictable whether it will succeed or fail. More investigation needed.

**Comment by eks-ebi** [comment]
After another test, it seems that the "Approve associations" bulk operation *does* work but takes a long time. The problems are:
* after clicking the button, there is no clear indication that the process has started, so curators do not know that it is working. Some sort of message should be displayed to improve the user experience.
* after clicking the button, it remains possible to click it again - and clicking again seems to interrupt the process and cause an error. The button should be disabled while the operation is in progress to avoid this.

**Comment by ljwh2** [comment]
@eks-ebi seems like this is working now, I will close ticket but please feed back if any other issues occur.
